### ISSUE TITLE
Reproducibility: NPZ generation, HCP-YA splits, and paper/checkpoint configuration alignment

### ISSUE THREAD

**Issue by redwangwangwang** [gustavochau/D-RoPE#2]
# Reproducibility: NPZ generation, HCP-YA splits, and paper/checkpoint configuration alignment

Hello, I am preparing HCP S1200 (2017) preprocessed diffusion data to reproduce the MAE pretraining and downstream experiments in the paper.

I reviewed commit [`7cb7bcc`](https://github.com/gustavochau/D-RoPE/commit/7cb7bcc1f7e6ea8635166ea92e5008361fe7f45a), the [paper and supplement](https://arxiv.org/html/2603.25977v1), the released [Hugging Face checkpoint/config/requirements](https://huggingface.co/gchau1/Drope/tree/fbad151c0482445326e63f45904e7e914b1c7beb), and the [DTI preprocessing pipeline cited as reference 1](https://doi.org/10.5281/zenodo.17503171). I understand that the README promises partial replication. Could you help identify or release the remaining code and settings needed for an end-to-end reproduction?

### 1. D-RoPE-specific preprocessing and NPZ generation

The separate DTI pipeline already provides correction, registration and tensor fitting. However, I could not find the stage that produces:

```text
<subject>/cubes_DTI_alldirdiv30_dipy/input_sample_<x>_<y>.npz
```

The [dataset loader](https://github.com/gustavochau/D-RoPE/blob/7cb7bcc1f7e6ea8635166ea92e5008361fe7f45a/utilities/data_utils.py#L126) expects `cubes`, `positions`, `sel_directions`, and `bvals` in these files. Could you share the packing script and paper-specific preprocessing configuration, including:

- How already-corrected HCP-YA NIfTI/bval/bvec files connect to the cited pipeline, the final MNI grid/cropping, and the gradient-coordinate convention after registration.
- The b=0 selection/averaging rule, normalization denominator, clipping limits, and handling of zero/invalid values.
- Array shapes, axis order and dtypes; the meanings of the two filename indices and `DTI_alldirdiv30`; direction grouping/order, sampling seeds, slab stride/overlap, and compression.
- How those files support the downstream `192×224×56` crop, and how auxiliary files such as `dti_mni_dipy/dipy_fa.nii.gz`, used by reconstruction, are generated/mapped.

A small synthetic NPZ example with expected shapes, plus a command that converts one locally obtained subject, would be very helpful.

### 2. Exact cohorts, splits and metadata mapping

The [pretraining entry point](https://github.com/gustavochau/D-RoPE/blob/7cb7bcc1f7e6ea8635166ea92e5008361fe7f45a/pretraining/run_training_mae_drope.py#L39) requires `train_subject_list` and `val_subject_list`; evaluation also requires `test_subject_list`. I found the HCP-A/HCP-D and ADNI downstream folds, but not the HCP-YA lists for the paper's 1,065-subject 70/15/15 split. Could you provide these lists or the exact selection/splitting script and seed, and confirm the HCP-YA release used?

There is also a split discrepancy to clarify: the published HCP-A fold 0 contains [354 training](https://github.com/gustavochau/D-RoPE/blob/7cb7bcc1f7e6ea8635166ea92e5008361fe7f45a/splits/hcp/train_subject_list_ag_0), [177 validation](https://github.com/gustavochau/D-RoPE/blob/7cb7bcc1f7e6ea8635166ea92e5008361fe7f45a/splits/hcp/val_subject_list_ag_0) and [177 test](https://github.com/gustavochau/D-RoPE/blob/7cb7bcc1f7e6ea8635166ea92e5008361fe7f45a/splits/hcp/test_subject_list_ag_0) subjects (50/25/25), whereas Section 4.3 states 30/35/35 for HCP-A/HCP-D. Which split produced the reported results?

The metadata CSVs are present, but the config files reference different external filenames. A mapping to the released CSVs, with the cohort filtering and normalization procedure, would make the fine-tuning scripts reproducible.

### 3. A canonical configuration matching the paper and checkpoint

The checked-in [pretraining JSON](https://github.com/gustavochau/D-RoPE/blob/7cb7bcc1f7e6ea8635166ea92e5008361fe7f45a/pretraining/pretraining_config_drope_conv.json) differs from Section 4.1/Table 5:

| Setting | Paper | GitHub JSON |
| --- | --- | --- |
| Patch size | 8×8×4 | 16×16×4 |
| Embedding dimension | 384 | 768 |
| Attention heads | 3 | 6 |
| Transformer decoder layers | 3 | 4 |
| Spatial masking ratio | 0.75 | 0.80 |

The Hugging Face YAML agrees with the paper on dimension/heads/decoder depth, but records spatial masking 0.70 and batch size 2, versus 0.75 and 4 in the paper; patch size is not recorded there. Could you provide a runnable training config and checkpoint-loading example, including whether batch size is per GPU or global?

A few concrete source/config differences also need clarification:

- The JSON sets `convolutional_final_layer` (singular), but the [parser](https://github.com/gustavochau/D-RoPE/blob/7cb7bcc1f7e6ea8635166ea92e5008361fe7f45a/pretraining/run_training_mae_drope.py#L84) reads `convolutional_final_layers` (plural). Evaluating the configuration function in isolation with the shipped JSON produces `convolutional_final_layers = {"value": false}`, disabling the convolutional decoder.
- The CLI defaults to `pretraining_config.json`, which is absent; an explicit documented launch command/config would avoid the fallback.
- For `mask_type=2`, [`forward_mask`](https://github.com/gustavochau/D-RoPE/blob/7cb7bcc1f7e6ea8635166ea92e5008361fe7f45a/models/mae_mixed_drope_conv.py#L313) switches masking on each forward call, while Section 4.1 describes switching per epoch. Which schedule was used?
- [Age fine-tuning](https://github.com/gustavochau/D-RoPE/blob/7cb7bcc1f7e6ea8635166ea92e5008361fe7f45a/downstream/age/Finetune_MAE_age.py#L298) applies LoRA, while Table 6 says no LoRA for age prediction. Which version should be used for that result?

### 4. Regenerating latents and the complete evaluation results

The released latent files and probing notebooks are useful. To regenerate the features from imaging, however, [`downstream/get_latents.py`](https://github.com/gustavochau/D-RoPE/blob/7cb7bcc1f7e6ea8635166ea92e5008361fe7f45a/downstream/get_latents.py#L65) still initializes a 768-dimensional, 6-head model with four decoder layers, unlike the released checkpoint configuration. Also, `downstream/aggregate_latents.py` is mentioned in the README but absent from the tree. Could you share checkpoint-compatible extraction and subject-level aggregation code, including the ADNI path and reproducible 12-direction selection?

For full table reproduction, I could not find a complete driver for the final FID calculation/aggregation, reconstruction-derived FA/MD errors, or all supervised/handcrafted baselines. Could you point to those scripts and their settings, or clarify which comparisons are outside the planned code release?

Finally, the [PSNR function](https://github.com/gustavochau/D-RoPE/blob/7cb7bcc1f7e6ea8635166ea92e5008361fe7f45a/pretraining/evaluation/run_psnr_ssim_evaluation.py#L15) divides by `sqrt(mse)`, rather than `mse`, and the reported SSIM code calls `ms_ssim`. An isolated check with reference 1.0, prediction 0.9 and data range 1.0 gives 10 dB with the current PSNR function, versus 20 dB with the standard definition. Could you confirm the precise metrics, masking and aggregation used for Table 1?

These observations come from source/artifact inspection and isolated configuration/metric checks; I have not completed a training run. My immediate priorities are the one-subject NIfTI→NPZ example, HCP-YA split files, and a configuration compatible with the released checkpoint. Those would make it possible to validate the pipeline before preparing the full dataset. Thank you.
